src/main/java/sg/christopher/duke/Duke.java

```diff
@@ -0,0 +1,273 @@
+package sg.christopher.duke;
+
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import sg.christopher.duke.entities.Deadline;
+import sg.christopher.duke.entities.Event;
+import sg.christopher.duke.entities.Task;
+import sg.christopher.duke.entities.Todo;
+import sg.christopher.duke.io.DataManager;
+import sg.christopher.duke.ui.MainWindow;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Duke {
+
+    private static List<Task> savedItems = loadSavedItems();
+
+    private MainWindow mainWindowController;
+    private static Stage stage;
+
+    public void setMainWindowController(MainWindow mw) {
+        mainWindowController = mw;
+    }
+
+    public void setStage(Stage stage) {
+        Duke.stage = stage;
+    }
+
+    private void dukePrint(String message) {
+        mainWindowController.printDukeMessage(message);
+    }
+
+    public void printWelcome() {
+        dukePrint("Hello I'm Duke! What can I do for you?");
+    }
+
+    /**
+     * Hydrates the task data store with saved data from disk.
+     *
+     * @return list of tasks that are saved on disk
+     */
+    public static List<Task> loadSavedItems() {
+        List<Task> saved = DataManager.readList();
+        return saved != null ? saved : new ArrayList<>();
+    }
+
+    /**
+     * Saves a task to memory and writes it to disk.
+     *
+     * @param task the task to be saved
+     */
+    public static void saveItem(Task task) {
+        savedItems.add(task);
+        DataManager.writeList(savedItems);
+    }
+
+    /**
+     * Loads saved tasks from disk, and get a specific task using its index.
+     *
+     * @param index index of the task to be retrieved
+     * @return the task at that index
+     */
+    public static Task loadItem(int index) {
+        savedItems = DataManager.readList();
+        return savedItems.get(index);
+    }
+
+    /**
+     * Removes saved tasks from disk, using its index.
+     *
+     * @param index index of the task to be removed
+     * @return task that is removed
+     */
+    public static Task removeItem(int index) {
+        Task removed = savedItems.remove(index);
+        DataManager.writeList(savedItems);
+        return removed;
+    }
+
+    private static String doneHandler(String userInput) {
+        int taskNo;
+        try {
+            taskNo = Integer.parseInt(userInput.split(" ")[1]);
+        } catch (ArrayIndexOutOfBoundsException aioobe) {
+            return "ERROR: No task no. found. Did you input the task no. of the task you'd like to mark as done?";
+        } catch (NumberFormatException nfe) {
+            return "ERROR: Unrecognized task. Please input the task no. of the task you'd like to mark as done.";
```

Comment:
Would be good to encapsulate errors into a DukeException Error class. Perhaps even to store a Enum for generic messages like: 
```
"You input: %s\n
Unrecognized task. Please input the task no. of the task you'd like to mark as done."
```